src/lazy/text/buffer.rs

```diff
@@ -1169,6 +1175,13 @@ impl<'data> TextBufferView<'data> {
                 contains_escaped_chars = true;
                 continue;
             }
+            if *byte == b'\r' {
+                // If the text contains an unescaped carriage return, we may need to normalize it.
+                // In some narrow cases, setting this flag to true may result in a sanitization buffer
+                // being allocated when it isn't strictly necessary.
+                contains_escaped_chars = true;
+                continue;
+            }
```

Comment:
🗺️ In long-form clobs and long-form strings, we need to normalize unescaped `\r` and `\r\n` to `\n`. This throws the naming off a bit; `contains_escapes` should really be something like `requires_substitutions`. However, I think `escapes` is a more obvious/suggestive name. Open to input here; I left it as-is because a consistent rename across usages/modules would touch a lot of lines and I'd rather do it in another PR.